src/types/middleware.ts

```diff
@@ -0,0 +1,28 @@
+import { EffectScope } from 'vue'
+import { MaybePromise } from '@/types/utilities'
+
+type Route = {
+  name: string,
+  params: Record<string, unknown>,
+  query: unknown,
+  hash: string,
+}
+
+interface RouteState {
+
+}
+
+interface RouteRejection {
+  NotFound: unknown,
+}
+
+type RouteReject = (type?: keyof RouteRejection) => void
+
+type MiddleWareExtras = {
```

Comment:
Should we provide the router here even though it won't have types?

Reply:
From my testing typescript at least seems happy if I just import a router into my middleware file and use it directly

Reply:
I believe when a project that implements this is built, both the routes, the router, and the middleware definitions end up in the same bundle. And since router is referenced/invoked in a function. There is no circular dependency/import. So as long as typescript is happy with this I think we're safe to say that importing your router and using it in middleware is recommended. But I could be easily convinced that providing the router as an "extra" would be beneficial. Maybe wait until someone asks for it

Reply:
I like the idea of including in extras. Even if we suggest and users get a better experience from importing, it's nice to provide an alternative syntax for users who are more comfortable not importing. That also would allow developers to write middleware that's agnostic of where or how to import a local router

Reply:
Agreed. Will update.